### ISSUE TITLE
--reset resets all deployments even when --tags are used

### ISSUE THREAD

**Issue by radeksvarz** [wighawag/hardhat-deploy#300]
Having a sequence of deploy scripts I wanted to redeploy only using 1 of them (redeploying 1 contract).

I expected the --reset would delete only those deployments, which are related to the given --tags.

However it deleted all deployments.

I understand this might be the design decision. Is there any work around in such case?

**Comment by wighawag** [comment]
tags are not known post deployment
actually tags are not necessarily tied to deployments, so --reset has to delete all

I guess you could have a script that delete contract you know are using which tags

**Comment by radeksvarz** [comment]
I found the workaround: 
 - delete contract(s) to be redeployed from .migrations.json
 - no --reset is needed as the deployments files are to be overwritten

Regarding --rest I suggest to update help to make it clear:
`whether to delete all deployments files first`

**Comment by jmendiola222** [comment]
Agree, I expected the same (only related Tags deployments to be reset). Even if the workaround does works, seems hacky, I would suggest renaming `--reset` to `reset-all` and/or add parameters to it, like: `--reset *` for all and `--reset NAME_1,NAME_2` for specific components.

**Comment by wighawag** [comment]
> Agree, I expected the same (only related Tags deployments to be reset). Even if the workaround does works, seems hacky, I would suggest renaming `--reset` to `reset-all` and/or add parameters to it, like: `--reset *` for all and `--reset NAME_1,NAME_2` for specific components.

--reset NAME cannot work as mentioned in my comment above

It would be good to have confirmation stating that happens though
